src/features/device-state/routes/state-get-v3.ts

```diff
@@ -334,20 +340,25 @@ const getDevice = async (req: Request, uuid: string) => {
 	return device;
 };
 
-const getConfig = (device: AnyObject) => {
+export const getConfig = (
+	device: AnyObject | undefined,
+	application: AnyObject = device?.belongs_to__application[0],
+) => {
 	const config: Dictionary<string> = {};
 
 	// add any app-specific config values...
-	const userAppFromApi: AnyObject = device.belongs_to__application[0];
+
 	varListInsert(
-		userAppFromApi.application_config_variable,
+		application?.application_config_variable ?? [],
 		config,
 		rejectUiConfig,
 	);
 
 	// override with device-specific values...
-	varListInsert(device.device_config_variable, config, rejectUiConfig);
-	filterDeviceConfig(config, device.os_version);
+	if (device) {
+		varListInsert(device.device_config_variable, config, rejectUiConfig);
+		filterDeviceConfig(config, device.os_version);
```

Comment:
Should we ask the fleet config to pass in an os version? As from the comment on `filterDeviceConfig` it seems important that it is run

Reply:
@Page- I see what you mean. On first sight I'd argue that the image that goes through this process is for devices greater than the os version constraint represented in the filterDeviceConfig. Hence, this filter can be applied always?

@aethernet Please add your insights on this for better clarity.

Reply:
As the filter only remove one var (`configVars.RESIN_HOST_LOG_TO_DISPLAY`) if OS is below v2 I think we're safe. 

From a preloading pov, we'll never preload v1 OS, so we're good (to be more precise preloading requires `overlay2` I don't remember the minimal os version shipping with an overlay2 engine but far above v2.

Reply:
@aethernet The check is for os semver >= 2.0.0 This would basically mean we always have to clean this up

Reply:
indeed sorry misread the sign, it should be always applied